features/yield-calculator/YieldCalculator.tsx

```diff
@@ -0,0 +1,120 @@
+import { Box, TextField, Button, Typography } from "@material-ui/core";
+import { FormEvent, useState } from "react";
+import styled from "styled-components";
+import EmpState from "../../containers/EmpState";
+
+const Container = styled.div`
+  padding: 1rem;
+  border: 1px solid #434343;
+`;
+
+const FormInput = styled.div`
+  margin-top: 20px;
+  margin-bottom: 20px;
+`;
+
+const MS_TO_S = 1000;
+const S_TO_DAYS = 60 * 60 * 24;
+const DAYS_TO_YEAR = 365;
+
+const YieldCalculator = () => {
+  const { empState } = EmpState.useContainer();
+  const [tokenPrice, setTokenPrice] = useState<number | null>(0.9575);
+
+  const expirationTimestamp = empState.expirationTimestamp;
+
+  const calculateDaysToExpiry = () => {
+    if (expirationTimestamp) {
+      const currentTimestamp = Math.round(Date.now() / MS_TO_S);
+      const secondsToExpiry = expirationTimestamp.toNumber() - currentTimestamp;
+
+      return Math.round(secondsToExpiry / S_TO_DAYS);
+    } else {
+      return null;
+    }
+  };
+  const [daysToExpiry, setDaysToExpiry] = useState<number>(
+    calculateDaysToExpiry() || 30
+  );
+
+  const calculateYield = () => {
+    if (!tokenPrice || tokenPrice <= 0) {
+      return null;
+    }
+    if (!daysToExpiry || daysToExpiry <= 0) {
+      return null;
+    }
+    const yieldPerUnit =
+      Math.pow(1 / tokenPrice, 1 / (daysToExpiry / DAYS_TO_YEAR)) - 1;
```

Comment:
nit: might be nice to have a comment briefly explaining the math here -- I'm having trouble intuiting the logic :)

Reply:
Done